docs/design_principles.txt

```diff
@@ -16,29 +16,86 @@ Many of these will be described with anti-examples ("don't").  I'd prefer to
+expressitivity so we're not going to try.  (TODO: explain why exceptions are
+okay / when they're okay.)
+
+This benefits humans, who spend more time reading code than writing it, and
+more effort debugging.  A human can read function signatures and know
+whether their bug may be or won't be within that function.
+
+It also benefits tools, specifically the compiler.  By making effects clear
+the compiler can perform more aggressive optimisations such as reordering or
+parallelising code.
 
 === Familiarity
 
 There are two aspects to familiarity.  One is generally using syntax that'll
-be more familiar to a majority of programmers in 2019.  We're assuming
-people coming to Plasma have at least 2 years experience programming and
-they may be "functionally curious".  Syntax for functions follows the
-popular curly-brace syntax of C-like languages.
+be more familiar to a majority of programmers in 2021.  We're assuming
+people coming to Plasma have at least two years experience programming and
+they may be "functionally curious".  By being familiar where we can it makes
+learning Plasma easier, and people can spend more of their energy learning
+the parts of Plasma that are different (usually by necessity).
+
+We make a number of choices about syntax that will be more familiar to most
+programmers.  For example Functions and blocks use curly-brace syntax of
+C-like languages and the body of a function or block is a series of
+statements.
 
 Likewise we use terminology and names that are going to be more familiar.
 What Haskell calls "Functor" we shall call "Mappable".  We know this
-isn't as accurate as "Functor", but we believe that it being more familiar to
-more people is a greater benefit than this level of accuracy.
+isn't as accurate as "Functor", but we're willing to loose some of that
+accuracy for more familiarity for more people.  Documentation will usually
+explain these kinds of choices, eg: "If you've used Haskell you may be used
+to calling these Functors, which is more accurate".  Which also makes it
+clear to people with that background exactly what they're looking at.
 
-The second aspect of familiarity is that something may not be familiar to 
-most people.  For example Plasma's syntax for ADTs borrows from Haskell,
-This syntax may be unfamiliar to the majority, but it's better to be
-familiar to those who have seen Haskell than no-one at all.
+Sometimes something is familiar to a smaller group of people.  Like ADTs, we
+use the Haskell syntax for ADTs because that's the syntax that's familiar to
+the largest group of people.
 
 Likewise some concepts have no familiar meaning (eg Monad).  We carefully
-weigh whether to include that concept (we will not include GADTs) and it is
-useful enough to include (Monads) we de-emphasise it.
+weigh whether to include that concept at all.  For example monads are very
+useful so we will include but de-emphasise them.  While GADTs are not
+generally useful so we will not include them.
```

Comment:
I trust you're preparing the bulwarks against the horde of Haskell fanatics who will be here to screech that yes, indeed, GADTs are generally useful, hence the "G" in the name.  :D

I'd suggest instead, "GADTs are more specialist in their use cases so…"

Reply:
Yeah, I had assumed I would underline "generally" in that sentence when they showed up.  But that I would be unable to convince them of my position regardless of what I said.  In any cause, I like your wording better.